### ISSUE TITLE
🐛 BUG: [TSX] Props silently degrade to `Record<string, any>` past a 1-byte threshold in total comment mass

### ISSUE THREAD

**Issue by victormelo-del** [withastro/compiler#1169]
### What version of `@astrojs/compiler` are you using?

2.13.1 (via `astro@7.2.2`, `@astrojs/check@0.9.10`, `typescript@6.0.3`)

### What package manager are you using?

npm

### What operating system are you using?

macOS 15 (Darwin 25.5.0, Apple Silicon)

### Describe the Bug

`convertToTSX` silently stops typing component props once a file crosses a **1-byte threshold in total comment mass**. Past the cliff it emits

```ts
export default function Repro__AstroComponent_(_props: Record<string, any>): any {}
```

instead of `_props: Props`, **and drops the whole `declare const Astro: Readonly<import('astro').AstroGlobal<Props, ...>>` block**. `diagnostics` is `[]` in both cases — the compiler reports nothing.

Downstream, `astro check` surfaces this as:

```
src/layouts/BaseLayout.astro:40:11 - error ts(6196): 'Props' is declared but never used.
```

which points at the one thing in the file that is provably correct. The obvious "fix" — deleting `interface Props` to silence the diagnostic — removes prop typing from the component for good. That is what makes this worth reporting: the visible symptom is a misleading diagnostic, but the actual damage is **prop types silently degrading to `any`** in a file that still compiles and still ships.

This looks related to #699 (same `_props: Record<string, any>` symptom, different trigger, fixed in 0.31.x).

### The trigger, bisected

Two ingredients, both necessary in our file; neither is sufficient on its own in a synthetic file (see "what I could not reduce" below).

**1. A tag name written inside a frontmatter block comment.** With `<main>` in a `/* ... */` comment in the frontmatter, the file gets a **finite comment budget**. Rewriting it as `main` (no angle brackets) makes the budget effectively unbounded — I padded a template comment to 40.000 B with no failure.

**2. Total comment mass.** With ingredient 1 present, growing **any** template comment by one byte past the threshold flips the output. Measured on our real 5.791 B layout file:

| variation | result |
| --- | --- |
| comment text content (accents, backticks, tags) | irrelevant — `xxxx` fails identically |
| +41 B spread over 6 short lines | ok |
| +155 B over 2 lines | fails |
| grow comment A **and** shrink comment B by the same amount | ok |
| +121 B of **real markup** (a long attribute value) | ok |

So it is not file size and not any particular character: it is the summed byte length of comments. Binary search on the real file: largest passing file **5.872 B**, smallest failing **5.873 B**.

### Link to Minimal Reproducible Example

Self-contained, no dependencies beyond the compiler. Save as `Repro.astro` (**986 bytes UTF-8 / 971 UTF-16 code units**) and run the script below.

<details>
<summary><code>Repro.astro</code> — 986 bytes, fails</summary>

```astro
---
/*
  Recebe o pacote de idioma inteiro em vez de decidir idioma por conta própria.
  Quem escolhe é a página, no caminho do import. O layout nunca precisa saber
  traduzir nada, e por isso não tem como traduzir errado.
  Landmarks: header > nav, main, footer > nav. O <main> tem id fixo porque o
  skip link aponta para ele.
*/
interface Props {
}
const {
} = Astro.props;
/** Alvo do skip link, no idioma desta página. Precisa casar com o id do <main>. */
---
    <meta charset="utf-8" />
    {/*
      Favicon PROVISÓRIO: as letras "DC" (public/favicon.svg, 568 B, 0 JS). A
      Dacoder tem símbolo de marca e vai enviá-lo — quando chegar, ele
      SUBSTITUI o arquivo, não convive com as letras. Um único rel="icon" em
      SVG: Chrome, Firefox, Safari e Edge atuais o aceitam, e .ico de 15 KB
      para navegador que não usamos como referência é peso sem leitor.
      xxxxxxxxxxxxxxxxx
    */}
      href="/fontes/archivo-latin-var-v25.woff2"
      as="font"```

</details>

```js
import { convertToTSX } from '@astrojs/compiler';
import { readFileSync } from 'node:fs';

const src = readFileSync('Repro.astro', 'utf8');
const body = (s) => s.split('//# sourceMappingURL=')[0];

const fails = await convertToTSX(src, { filename: 'Repro.astro' });
// remove exactly ONE character from the padding run of x's
const ok = await convertToTSX(src.replace('xx\n', 'x\n'), { filename: 'Repro.astro' });

console.log('986 B:', body(fails.code).match(/_props: [^)]*/)[0], '| diagnostics:', fails.diagnostics.length);
console.log('985 B:', body(ok.code).match(/_props: [^)]*/)[0],    '| diagnostics:', ok.diagnostics.length);
```

Output:

```
986 B: _props: Record<string, any> | diagnostics: 0
985 B: _props: Props              | diagnostics: 0
```

**Deleting a single `x` from a comment restores prop typing.** The `declare const Astro` block disappears and reappears with it.

### What I could not reduce

Honest limitation: I could not build this up from scratch. Synthetic files with each ingredient in isolation — a tag inside a frontmatter block comment, a tag inside a JSDoc comment, both together, a full `<!doctype html>` document, 1 to 4 separate template comments, comments up to 8.000 B — all preserve `_props: Props`. The 986 B file above was produced by automated delta-debugging **down from** a real 5.791 B layout, using a contrast-preserving predicate (a candidate only counts if it fails **with** the padding and passes **without** it) plus a well-formedness check (exactly two `---` delimiters, balanced comment tokens, frontmatter parsing clean under the TypeScript parser).

The prose inside the comments is real text from our codebase, in Portuguese; it is filler as far as the bug is concerned — replacing it with `x` of the same length reproduces identically.

### Why the byte counts matter operationally

Before we found this, our layout was sitting **81 bytes** from the cliff, with no way to know. Any future comment — one sentence — would have silently dropped prop typing across every page of the site, and the error message would have sent whoever hit it to the wrong file.
